DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/matmul/src/matmul_demo.cpp

```diff
@@ -0,0 +1,225 @@
+#include <iomanip>
+#include <iostream>
+
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/sycl.hpp>
+
+#include "exception_handler.hpp"
+#include "matmul.hpp"
+
+// Fills a matrix with random numbers within the range [l_bound, u_bound).
+void FillRand(std::vector<float> &m_matrix, int l_bound, int u_bound,
+              int elements) {
+  for (int element = 0; element < elements; element++) {
+    m_matrix[element] =
+        static_cast<float>(rand()) /
+            (static_cast<float>((RAND_MAX) / (u_bound - l_bound))) +
+        l_bound;
+  }
+}
+
+// Compares num_matrices pairs of matrices; returns true iff they are equal
+// given a tolerated error bound.
+bool EqualMat(std::vector<float> &c_matrix, std::vector<float> &c_reference,
+              int rows, int cols, int num_matrices) {
+  int matsize = rows * cols;
+  bool passed = true;
+
+  // Floating-point error threshold value
+  constexpr float kEpsilon = 0.01f;
+
+  for (int matrix_idx = 0; matrix_idx < num_matrices; matrix_idx++) {
+    for (int col = 0; col < cols; col++) {
+      for (int row = 0; row < rows; row++) {
+        int idx = matrix_idx * matsize + col * rows + row;
+        if (abs(c_matrix[idx] - c_reference[idx]) > kEpsilon) {
+          passed = false;
+#if DEBUG
+          std::cout << "Error: C[" << col << "][" << row << "] = "
+                    << c_matrix[idx]
+                    << " but REF[" << col << "][" << row << "] = "
+                    << c_reference[idx] << std::endl;
+#endif
+        }
+        if (!std::isfinite(c_matrix[idx])) {
+          passed = false;
+#if DEBUG
+          std::cout << "C[" << col << "][" << row << "] = " << c_matrix[idx]
+                    << " is not finite" << std::endl;
+#endif
+        }
+      }
+    }
+  }
+  return passed;
+}
+
+// Output a matrix to the screen (assumes column-major format).
+void PrintMat(std::vector<float> &m_matrix, int rows, int cols) {
+  for (int row = 0; row < rows; row++) {
+    for (int col = 0; col < cols; col++) {
+      std::cout << std::fixed;
```

Comment:
This will make Coverity complain, have a look at this PR: https://github.com/oneapi-src/oneAPI-samples/commit/aa5b05fc3147f4b2ae012e8e2157780c50c684c9